### PR TITLE
Add password reset feature

### PR DESCRIPTION
**Pull request by yliu0715**

### CODE REVIEW

**Comment by yliu0715** [comment]
Idk why gts fix did not fix some of the lines

**Comment by yliu0715** [comment]
@melvinheks They were all just calling client.execute()s

**Comment by melvinheks** [comment]
What if you called the wrong query

File: src/web/server/app/utils/mailer.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import nodemailer from 'nodemailer';
+import { EMAIL, EMAIL_PASSWORD, REACT_SERVER_IP } from './settings';
+
+export const sendEmail = async (recipient: string, token: string) => {
+  try {
+    const transporter = nodemailer.createTransport({
+      service: 'Gmail',
```
Comment (melvinheks) [inline src/web/server/app/utils/mailer.ts:7]: This wouldn't work if EMAIL isn't gmail right?

File: src/web/server/app/services/auth.ts [hunk 2]
```diff
@@ -153,3 +160,75 @@ export const approveRegistration = (
...
+export const storeResetToken = (token: string, userId: string) => {
+  client.execute(INSERT_PASSWORDRESETTOKENS, [token, userId], {
+    prepare: true,
+  });
+};
+
+/**
+ * Retrieve the userId and email using username as lookup
+ * @params username - The username of the user
+ */
```
Comment (melvinheks) [inline src/web/server/app/services/auth.ts:178]: Add @returns to functions that return

File: src/web/server/app/utils/mailer.ts (hunk 1 above)
Comment (yliu0715) [reply src/web/server/app/utils/mailer.ts:7]: Guess not.

File: src/web/server/app/services/auth.ts [hunk 3]
```diff
@@ -153,3 +160,78 @@ export const approveRegistration = (
...
+  const result = await client.execute(SELECT_PASSWORDRESETTOKENS, [token], {
+    prepare: true,
+  });
+  return result.rows.length === 1;
+};
+
+/**
+ * Retrieve the associated user ID using the reset token
+ * @params token - The password reset token
+ * @returns any - The id of the retrieved user
```
Comment (melvinheks) [inline src/web/server/app/services/auth.ts:201]: Isn't the type a string?

Comment (yliu0715) [reply src/web/server/app/services/auth.ts:201]: I thought so too, but when I hover my cursor over the return value, it said `any`, so I wanted to be consistent with what it said.

Comment (yliu0715) [reply src/web/server/app/services/auth.ts:201]: @melvinheks So should I put string or any?

Comment (melvinheks) [reply src/web/server/app/services/auth.ts:201]: string probably

**Pull request merged**